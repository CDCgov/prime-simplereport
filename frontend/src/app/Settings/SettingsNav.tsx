import React from "react";

import { LinkWithQuery } from "../commonComponents/LinkWithQuery";
import { NewFeatureTag } from "../commonComponents/NewFeatureTag";

const SettingsNav = () => {
  return (
    <nav className="prime-secondary-nav" aria-label="Secondary navigation">
      <ul className="usa-nav__secondary-links prime-nav">
        {process.env.REACT_APP_USER_SETTINGS_ENABLED === "true" ? (
          <li className="usa-nav__secondary-item">
            <LinkWithQuery
              to={`/settings`}
              activeClassName="active"
              exact={true}
            >
              Manage Users
            </LinkWithQuery>
          </li>
        ) : null}
        <li className="usa-nav__secondary-item">
          <LinkWithQuery to={`/settings/facilities`} activeClassName="active">
            Manage Facilities
          </LinkWithQuery>
        </li>
        <li className="usa-nav__secondary-item">
          <LinkWithQuery to={`/settings/organization`} activeClassName="active">
            Manage Organization
          </LinkWithQuery>
        </li>
        <li className="usa-nav__secondary-item">
          <LinkWithQuery
            to={`/settings/self-registration`}
            activeClassName="active"
          >
            Patient self-registration{" "}
            <NewFeatureTag feature="selfRegistration" />
          </LinkWithQuery>
        </li>
      </ul>
    </nav>
  );
};

export default SettingsNav;
