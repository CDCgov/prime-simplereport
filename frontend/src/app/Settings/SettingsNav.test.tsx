import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import SettingsNav from "./SettingsNav";

describe("SettingsNav", () => {
  it("displays the nav order correctly and defaults to 'Manage Users'", () => {
    const { container } = render(
      <MemoryRouter>
        <SettingsNav />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
  describe("ManageFacility", () => {
    it("displays as active when viewing all facilities", () => {
      render(
        <MemoryRouter initialEntries={["/settings/facilities"]}>
          <SettingsNav />
        </MemoryRouter>
      );
      expect(screen.getByText("Manage facilities")).toHaveAttribute(
        "aria-current",
        "page"
      );
      expect(screen.getByText("Manage facilities")).toHaveClass("active", {
        exact: true,
      });
    });
    it("displays as active when viewing specific facility", () => {
      render(
        <MemoryRouter initialEntries={["/settings/facility/some-uuid"]}>
          <SettingsNav />
        </MemoryRouter>
      );
      expect(
        screen.getByText("Manage facilities").parentElement
      ).toHaveAttribute("aria-current", "page");
      expect(screen.getByText("Manage facilities")).toHaveClass("active", {
        exact: true,
      });
    });
    it("displays as active when adding new facility", () => {
      render(
        <MemoryRouter initialEntries={["/settings/add-facility/some-uuid"]}>
          <SettingsNav />
        </MemoryRouter>
      );
      expect(
        screen.getByText("Manage facilities").parentElement
      ).toHaveAttribute("aria-current", "page");
      expect(screen.getByText("Manage facilities")).toHaveClass("active", {
        exact: true,
      });
    });
  });
});
