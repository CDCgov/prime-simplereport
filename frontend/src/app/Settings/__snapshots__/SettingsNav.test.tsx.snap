// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsNav displays the nav order correctly and defaults to 'Manage Users' 1`] = `
<div>
  <nav
    aria-label="Secondary navigation"
    class="prime-secondary-nav"
  >
    <ul
      class="usa-nav__secondary-links prime-nav"
    >
      <li
        class="usa-nav__secondary-item"
      >
        <a
          href="/settings"
        >
          Manage Users
        </a>
      </li>
      <li
        class="usa-nav__secondary-item"
      >
        <a
          href="/settings/facilities"
        >
          Manage Facilities
        </a>
      </li>
      <li
        class="usa-nav__secondary-item"
      >
        <a
          href="/settings/organization"
        >
          Manage Organization
        </a>
      </li>
      <li
        class="usa-nav__secondary-item"
      >
        <a
          href="/settings/self-registration"
        >
          Patient self-registration
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
